### PR TITLE
[MRG] enforce that name be a required file in dataset_description.json

### LINKED ISSUE #278: name is a required field in the dataset_description.json
We are not enforcing that the name parameter is required. This causes an error with the validator.

### PR DESCRIPTION
**Pull request by teonbrooks**
closes #278

Thanks for contributing. If this is your first time,
make sure to read [contributing.md](https://github.com/mne-tools/mne-bids/blob/master/CONTRIBUTING.md)

PR Description
--------------

Describe your PR here

Merge checklist
---------------

Maintainer, please confirm the following before merging:

- [ ] All comments resolved
- [ ] This is not your own PR
- [x] All CIs are happy
- [x] PR title starts with [MRG]
- [x] [whats_new.rst](https://github.com/mne-tools/mne-bids/blob/master/doc/whats_new.rst) is updated
- [x] PR description includes phrase "closes <#issue-number>"
- [x] Commit history does not contain any merge commits

### CODE REVIEW

File: doc/whats_new.rst [hunk 1]
```diff
@@ -30,6 +30,7 @@ Bug
 
 API
 ~~~
+- Enforce that the name arg is a required field for :func`mne_bids.make_dataset_description`, by `Teon Brooks`_(`#279 <https://github.com/mne-tools/mne-bids/pull/279>`)
```
Comment (jasmainak) [inline doc/whats_new.rst]: I think this should go in the Bug section since this is not a backwards compatible change

**Comment by jasmainak** [comment]
why were none of our tests failing so far?

**Comment by teonbrooks** [comment]
it looks as though that function wasn't tested...

**Comment by jasmainak** [comment]
should we add a test then? to avoid regression in the future.

**Comment by teonbrooks** [comment]
@jasmainak the function `make_dataset_description` has no tests. just rebased and address the whats_new. unfortunately i don't have the time to make all the tests for this function. feel free to take it over

**Comment by sappelhoff** [comment]
ok, we forgot to merge it and now there are conflicts :)

see #342

**Pull request closed without merging**